src/main/java/io/xpring/Utils.java

```diff
@@ -68,6 +68,37 @@ public static String byteArrayToHex(byte [] bytes) {
         return data;
     }
 
+    /**
+     * Encode the given classic address and tag into an X-Address.
+     *
+     * @param classicAddress A classic address to encode.
+     * @param tag An optional tag to encode.
+     * @return A new X-Address if inputs were valid, otherwise undefined.
+     * @see <a href="https://xrpaddress.info/">https://xrpaddress.info/</a>
+     */
+    public static String encodeXAddress(String classicAddress, Long tag) {
+        try {
+            return javaScriptUtils.encodeXAddress(classicAddress, tag);
+        } catch (Exception exception) {
+            throw new RuntimeException(exception);
+        }
+    }
+
+    /**
+     * Decode a {@link ClassicAddress} from a given X-Address.
+     *
+     * @param xAddress The xAddress to decode.
+     * @return A {@link ClassicAddress} if the inputs were valid, otherwise null.
+     * @see <a href="https://xrpaddress.info/">https://xrpaddress.info/</a>
+     */
+    public static ClassicAddress decodeXAddress(String xAddress) {
```

Comment:
If `null` isn't a valid argument it may be worth using `Objects.requireNonNull(xAddress)` here.

Reply:
@keefertaylor One style paradigm I really like in Java is to treat `null` values as a bug. With proper unit testing, this can save a lot of pain over the long-term. My thinking here was inspired by the Google Guava engineers per this blog post: https://github.com/google/guava/wiki/UsingAndAvoidingNullExplained

Reply:
Fixed through the rest of the file as well.